app/src/main/java/com/github/se/polyfit/ui/components/selector/MealOccasionSelector.kt

```diff
@@ -0,0 +1,96 @@
+package com.github.se.polyfit.ui.components.selector
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.RadioButton
+import androidx.compose.material3.RadioButtonColors
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.github.se.polyfit.model.meal.MealOccasion
+import com.github.se.polyfit.ui.theme.PrimaryPurple
+import com.github.se.polyfit.ui.theme.PurpleGrey40
+import com.github.se.polyfit.ui.utils.titleCase
+
+@Composable
+fun MealOccasionSelector(
+    startOccasion: MealOccasion,
+    onConfirm: (MealOccasion) -> Unit,
+    modifier: Modifier = Modifier
+) {
+  var mealOccasion by remember { mutableStateOf(startOccasion) }
+
+  fun onClick(occasion: MealOccasion) {
+    mealOccasion = occasion
+    onConfirm(occasion)
+  }
+
+  Column(modifier = modifier.fillMaxWidth().testTag("MealOccasionSelector")) {
+    Text(
+        text = "Meal Occasion",
+        style = MaterialTheme.typography.titleLarge,
+        color = PurpleGrey40,
+        modifier = Modifier.padding(16.dp, 0.dp).testTag("Title"))
+    Row(
+        modifier = Modifier.fillMaxWidth().testTag("ButtonRow"),
+        horizontalArrangement = Arrangement.SpaceEvenly) {
+          Column(modifier = Modifier.testTag("LeftColumn")) {
```

Comment:
It might be a good idea to iterate over the elements of the enum instead of using hardcoded values. This could improve modularity if we decide to add or remove a MealOccasion in the future.

Reply:
It's a good point. I think as a counter argument, two things:
1. I think its highly unlikely for the enum to change. At worst the name would change, and the code wont compile and will need a quick change. I think there will only ever really be 4 options for this.
2. We have an "Other" enum, but that's basically just used to create a new meal and have it as an invalid state until a proper one is selected. If we iterate over the enums, we would have to hardcode to exclude it and this would become a burden in a different way if we change the order/add to the enums.

Reply:
I missed the fact that you did not include `MealOccasion.OTHER`; yes, your implementation is better suited to what I was thinking.